### PR TITLE
Remove existing tag code from Analytics

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #4702

I'm spinning my wheels a bit on the last failing E2E test (ignore the AMP one as I think it's just a timeout): https://github.com/google/site-kit-wp/runs/5223116819?check_suite_focus=true#step:14:249

I've not made any changes that I can see to that scenario, and looking at it in Puppeteer the relevant screen appears and the "re-fetch" button is clicked. Even when I ran it in interactive mode, the request wasn't fired. Normally I'd think that was because I messed up a component that does the re-fetching… but I haven't touched those components at all from what I can tell. 🤔 

I'll take another look tomorrow with fresh eyes, but I've spent a lot of time trying to debug that and feel like I might just be missing something obvious as there's a lot of code/changes in this PR.

Aside from that, though, I think this is all set for review and seems to work well.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by tofumatt** [comment]
Yeah, I mentioned the failing test in the PR summary—I went through it and can't see what's changed/why it's failing; I've sunk a lot of time into it but nothing jumps out at me, so I figured it'd be good to have someone else take a look 😅 

If it's still failing, could you take a look or see if something in the PR is the obvious cause of the failure? I've not changed  code related to that failing test as far as I know, so I'm a bit perplexed 🤔

**Comment by techanvil** [comment]
> Yeah, I mentioned the failing test in the PR summary—I went through it and can't see what's changed/why it's failing; I've sunk a lot of time into it but nothing jumps out at me, so I figured it'd be good to have someone else take a look sweat_smile
> 
> If it's still failing, could you take a look or see if something in the PR is the obvious cause of the failure? I've not changed code related to that failing test as far as I know, so I'm a bit perplexed thinking

Thanks @tofumatt, and sorry for not spotting your comment in the PR summary. I've taken a look, and, although it took a while, I have found the issue (a one-liner, as can be expected!). I've pushed the fix in this [commit](https://github.com/google/site-kit-wp/pull/4835/commits/44587e47fae78619ce8844d9a1c96f2df0247b9d).

File: assets/js/modules/analytics/datastore/accounts.js [hunk 1]
```diff
@@ -56,6 +56,7 @@ const fetchGetAccountsPropertiesProfilesStore = createFetchStore( {
 			'modules',
 			'analytics',
 			'accounts-properties-profiles',
+			null,
```
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/modules/analytics/datastore/accounts.js:59]: No problem! I must say, this is one of those bugs where I do miss working with TypeScript...

File: assets/js/modules/analytics/hooks/useExistingTagEffect.js [hunk 2]
```diff
@@ -28,97 +28,20 @@ import { useEffect } from '@wordpress/element';
...
-		return data;
-	} );
+	const hasExistingTag = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).hasExistingTag()
+	);
 
 	useEffect( () => {
-		if ( existingTag ) {
-			// Disable the plugin snippet
+		if ( hasExistingTag ) {
```
Comment (techanvil) [inline assets/js/modules/analytics/hooks/useExistingTagEffect.js:41]: Reading this point in the IB, it seems to suggest that `useExistingTagEffect` should disable the snippet if there is an existing tag _and_ the ID is the same as the currently configured property ID. However, I can see that you have updated `UseSnippetSwitch` to contain this logic. This seems like it's probably the right thing to do, and also the IB is a little ambiguous in its phrasing. So, I just want to highlight this and make sure this is the right way to go about things here.

> Remove all use of getTagPermission from useExistingTagEffect and drastically simplify it so that all that it does going forward is disable the plugin snippet if the existing tag is the same as the currently configured property ID

**Comment by techanvil** [comment]
Hey @tofumatt, I have noticed this PR is targeting `develop` but it should in fact be targeting `feature/existing-tag-simplification`, and as such will likely need a rebase onto the feature branch too.

**Comment by techanvil** [comment]
I've noticed that some of the related Storybook stories are now in a bit of an unusable state, however they should be reassessed once this branch has been rebased onto `feature/existing-tag-simplification`. These stories have both the dropdowns and button disabled:

**Analytics Module -> Settings -> No Tag, GTM property w/ access**
image

**Analytics Module -> Settings -> No Tag, GTM property w/o access**
image

**Analytics Module -> Setup -> No Tag, GTM property w/ access**
image

**Analytics Module -> Setup -> No Tag, GTM property w/o access**
image

**Review by techanvil — CHANGES_REQUESTED** [review]
Hi @tofumatt, nice work! This broadly looks good, with just a few issues that do/may need addressing, please see the included comments. Cheers

File: assets/js/modules/analytics/hooks/useExistingTagEffect.js (hunk 2 above)
Comment (tofumatt) [reply assets/js/modules/analytics/hooks/useExistingTagEffect.js:41]: Yeah, I thought this was the best place to put it but I did notice that. Sorry I forgot to mention it here 👍🏻

File: includes/Modules/Analytics.php [hunk 3]
```diff
@@ -700,25 +699,6 @@ function ( $metric_def ) {
...
-							array( 'status' => 400 )
-						);
-					}
-					$property_id = $data['propertyID'];
-					return array_merge(
-						array(
-							'accountID'  => '', // Set the accountID to be an empty string and let has_access_to_property handle determining actual ID.
-							'propertyID' => $property_id,
-						),
-						$this->has_access_to_property( $property_id )
```
[the remark this answers is not in the document]
Comment (tofumatt) [reply includes/Modules/Analytics.php:719]: Good catch, thanks! 👍🏻

File: assets/js/modules/analytics/hooks/useExistingTagEffect.js (hunk 2 above)
Comment (techanvil) [reply assets/js/modules/analytics/hooks/useExistingTagEffect.js:41]: Makes sense, and no problem!

**Comment by techanvil** [comment]
> I've noticed that some of the related Storybook stories are now in a bit of an unusable state, however they should be reassessed once this branch has been rebased onto `feature/existing-tag-simplification`. These stories have both the dropdowns and button disabled:

Good to see these are all sorted now!

**Review by techanvil — APPROVED** [review]
LGTM, nice one @tofumatt!

File: assets/js/components/notifications/SetupSuccessBannerNotification.js [hunk 4]
```diff
@@ -169,10 +173,6 @@ function SetupSuccessBannerNotification() {
 				}
 			}
 
-			if ( serviceSetupV2Enabled ) {
-				winData.description = '';
-			}
```
Comment (felixarntz) [inline assets/js/components/notifications/SetupSuccessBannerNotification.js:174]: The changes in this file seem to have gotten here accidentally? They seem unrelated.

File: assets/js/modules/analytics/components/common/UseSnippetSwitch.js [hunk 5]
```diff
@@ -52,7 +63,13 @@ export default function UseSnippetSwitch() {
 		);
 	}, [ useSnippet, setUseSnippet, viewContext ] );
 
-	if ( useSnippet === undefined ) {
+	useEffect( () => {
+		if ( toggleSwitchDisabled ) {
+			setUseSnippet( false );
+		}
+	}, [ setUseSnippet, toggleSwitchDisabled ] );
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/common/UseSnippetSwitch.js:70]: Can you explain this change? Why do we set the snippet to `false` here? That seems new and isn't specifically mentioned in the ACs (or IB) as far as I can tell.

File: assets/js/modules/analytics/components/common/UseSnippetSwitch.js [hunk 6]
```diff
@@ -66,7 +83,7 @@ export default function UseSnippetSwitch() {
 				checked={ useSnippet }
 				onClick={ onChange }
 				hideLabel={ false }
-				disabled={ ! canUseSnippet }
+				disabled={ ! canUseSnippet || toggleSwitchDisabled }
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/common/UseSnippetSwitch.js:86]: See above. It seems unnecessary to disable the snippet control here? In the setup we currently don't show it all right? So while we need to set `useSnippet` to `false` during setup if there's an existing tag, I'm not sure why it's necessary to disable the toggle. Can you clarify? Maybe I missed something from another conversation.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/DimensionTabs.js [hunk 7]
```diff
@@ -143,5 +149,6 @@ export default function DimensionTabs( { dimensionName, loaded } ) {
 
 DimensionTabs.propTypes = {
 	dimensionName: PropTypes.string.isRequired,
+	gatheringData: PropTypes.bool,
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/DimensionTabs.js:152]: The changes in this file seem unrelated?

File: assets/js/modules/analytics/hooks/useExistingTagEffect.js (hunk 2 above)
Comment (felixarntz) [reply assets/js/modules/analytics/hooks/useExistingTagEffect.js:41]: I'd like to follow up on this thread, as @techanvil mentioned above, the `useSnippet` setting should indeed only be set to `false` if the existing tag is also the same as the currently set property ID. So it looks to me that this hook here needs that update still.

Regarding doing that in `UseSnippetSwitch`, that seems problematic to me since that component isn't necessarily used, for example we only display it in Analytics settings right now, but not in the Analytics setup. The logic to conditionally hide the snippet should apply in either case though.

**Review by felixarntz — COMMENTED** [review]
@tofumatt I left a few comments for this PR, but it also seems that there are a bunch of unrelated changes included here. Can you double-check the base branch and your branch to make sure only the relevant changes are included?

File: assets/js/modules/analytics/components/common/UseSnippetSwitch.js (hunk 5 above)
Comment (tofumatt) [reply assets/js/modules/analytics/components/common/UseSnippetSwitch.js:70]: Yeah, this wasn't mentioned in the ACs, but if the switch to use our snippet is disabled, I thought it best to disable our snippet. Otherwise, if the snippet were disabled but the setting was previously set to `true`, then the user would be forced to output our snippet _and_ the existing snippet, with no way to disable ours because the snippet setting cannot be changed.

But actually I think this needs some further thought, you're right 👍🏻

**Comment by tofumatt** [comment]
Thanks for that review @felixarntz. I've actually updated the code so now the snippet is only disabled when the existing tag matches the propertyID set. I've added a test for this as well.

Because I made this change in the `useExistingTagEffect` hook, it works everywhere. The snippet code is set not to insert by virtue of setting the setting during setup. 👍🏻

File: assets/js/modules/analytics/hooks/useExistingTagEffect.js [hunk 8]
```diff
@@ -28,97 +28,40 @@ import { useEffect } from '@wordpress/element';
...
-			// GTM container has GA tag and user has access to it, force select it.
-			setAccountID( gtmAnalyticsAccountID );
-			selectProperty( gtmAnalyticsPropertyID );
-			// We no longer set `useSnippet` to false here as this is handled using a filter
-			// on the server side now so that Analytics will take over the snippet again
-			// if Tag Manager or its snippet is disabled.
+			// Disable the Analytics snippet if there is an existing tag that
+			// matches the currently selected property.
+			setUseSnippet( false );
+			setCanUseSnippet( false );
```
Comment (felixarntz) [inline assets/js/modules/analytics/hooks/useExistingTagEffect.js:58]: Why the `setCanUseSnippet` here? That is a different option to enforce the value of `useSnippet`. We don't set that today, and it shouldn't happen here either. For example, if the `UseSnippetSwitch` toggle is present in the UI, it shouldn't be disabled in such a case, we should only change its value (via `setUseSnippet`).

**Review by felixarntz — COMMENTED** [review]
@tofumatt A few follow-up comments here: We shouldn't need to touch `canUseSnippet`. Also, since the snippet placement now depends on the property selected, we also need to ensure we consider the case where the user switches from a property that matches the existing tag to a property that doesn't (in which case the snippet should be placed again).

File: assets/js/modules/analytics/hooks/useExistingTagEffect.js (hunk 8 above)
Comment (tofumatt) [reply assets/js/modules/analytics/hooks/useExistingTagEffect.js:58]: Ah, okay, I thought we would disable it only _if_ the property matched. Instead we'll just disable it, sounds good 👍🏻
